### ISSUE TITLE
can your software be used on a local machine

### ISSUE THREAD

**Issue by cindylanzao-design** [MeHelmy/princess#3]
We want to use your software to detect SV and SNV on ONT fasta file, however, we do not have a cluster, can your software be used on a local machine?If it could, how to revise the config.yaml file?

**Comment by MeHelmy** [comment]
Thank you for reaching out.
In short, Yes, it can run on your local machine using parameter `--NoCluster` or `-e` and it was tested on Linux.

But can you give more information e.g expected coverage, it is WGS data or you interested in one chromosome only, what is your machine configuration (RAMs, CPU, MAC, etc..), you want to call SV and SNVs phased or not? so I can give a more detailed answer.

~Best,
Medhat

**Comment by cindylanzao-design** [comment]
Hi,

Thanks for reaching us. We are now using it on a test data, in the future, we will want to test it on WGS data, and phasing is also what we want. The machine is a cloud machine, 16core CPU Ubuntu machine, over 64GB of memory.

Best,
Xinyue

发件人: Medhat [mailto:notifications@github.com]
发送时间: 2020年2月11日 11:35
收件人: MeHelmy/princess <princess@noreply.github.com>
抄送: lixinyue (H) <lixinyue12@huawei.com>; Author <author@noreply.github.com>
主题: Re: [MeHelmy/princess] can your software be used on a local machine (#3)


Thank you for reaching out.
In short, Yes, it can run on your local machine using parameter --NoCluster or -e and it was tested on Linux.

But can you give more information e.g expected coverage, it is WGS data or you interested in one chromosome only, what is your machine configuration (RAMs, CPU, MAC, etc..), you want to call SV and SNVs phased or not? so I can give a more detailed answer.

~Best,
Medhat

**Comment by cindylanzao-design** [comment]
By the way，the –e or –NoCluster parameter seems invalid, do you ever update the Princess?

Thanks

Best,
Xinyue

发件人: Medhat [mailto:notifications@github.com]
发送时间: 2020年2月11日 11:35
收件人: MeHelmy/princess <princess@noreply.github.com>
抄送: lixinyue (H) <lixinyue12@huawei.com>; Author <author@noreply.github.com>
主题: Re: [MeHelmy/princess] can your software be used on a local machine (#3)


Thank you for reaching out.
In short, Yes, it can run on your local machine using parameter --NoCluster or -e and it was tested on Linux.

But can you give more information e.g expected coverage, it is WGS data or you interested in one chromosome only, what is your machine configuration (RAMs, CPU, MAC, etc..), you want to call SV and SNVs phased or not? so I can give a more detailed answer.

~Best,
Medhat

**Comment by MeHelmy** [comment]
First I would like you to update Princess `git pull`, then you can see the `-e` parameter.

I tested Princess on Linux machine with 40G of RAMs and it runs fine (I used only chromosomes 22).

For `config.yaml` file:

The reference must be indexed, to do that use `samtools faidx your_refernce.fai`.

Then let us go to the `config.yaml`  to add the required information:
under the title `Information about reference` we will update these fields:

In this field `ref: []` add your reference version either 37 or 38: e.g `ref: ['37']`

In `references:` add the location of the reference: e.g `'37': "/home/user/somdir/genome.fa"`

AS we are just testing we will use chromosome 22 (you can use what you find interesting for you) 
to know what your chromosomes named like take a look at the first column in the indexed genome in our case it will be `/home/user/somdir/genome.fa.fai`

`chr_list:
  '37': ['22']` # leave free if you wish to call all chromosomes ex: `'37': []`.


So in the end, it will be:

```
ref: ['37']
references:
  '37': "/home/user/somdir/genome.fa"
  '38': "
chr_list:
  '37': ['22'] 
  '38': [] 

```

Now, let us say that you have a fasta file in `/home/user/somdir/sample.fasta` 

To test what shall run use:
`princess_location/princess/princess -c all  -s /home/user/somdir/sample.fasta -d ./analysis_37   -r ont -e  -npr`

```
-c    The command that will run
-s    One or more input file
-d   Where the output should go
-r    Data type
-e  Do not use the cluster configuration
```
Results will be something like:


> Building DAG of jobs...
> Job counts:
> 	count	jobs
> 	1	SV_SNP_compained
> 	1	add_rg
> 	1	all
> 	1	all_phased
> 	1	bam_2_tab
> 	1	bam_stat
> 	2	bgzip_file
> 	2	call_snps_chunk
> 	1	change_sample_name
> 	1	concat_chromosome
> 	1	gt
> 	4	index_bam
> 	1	merge_align
> 	2	minimap2
> 	1	partion_bam
> 	1	phase_sv
> 	1	phasing
> 	1	reads_stat
> 	1	sniffles
> 	1	snp_stat
> 	1	stat
> 	1	sv_stat
> 	2	vcf_index
> 	1	vcf_sort
> 	31  # this is the number of jobs that will run, it will change if we used more samples.

**If you take a look to the analysis_37 directory you will find new directories were created, Besides symbolic links to the input fasta files**
bin/
envs/
modules/
pictures/
scripts/

**Time to do the real run.**
My input file is a fastq file for chromosome 22
Reads: 173298
Bases: 681069807
Mean read length: 3930.0500121178547
Median: 1984.0
Max: 754862
Min: 173
N50: 7643

`princess/princess -c all  -s minimap_37_chr22.fastq  -d ./analysis_37   -r ont -e`


Building DAG of jobs...
Creating conda environment envs/princess_env.yaml...
Downloading and installing remote packages.
Environment for envs/princess_env.yaml created (location: .snakemake/conda/b929bbb8)  # Name of the conda environment will be different, and you may need to delete after you finish your analysis.
Creating conda environment envs/clair_env.yaml...
Downloading and installing remote packages.

This will result in (In your case will be different sample name)

Job 15: Running minimap2 , sample is: minimap_37_chr22.fastq

for each job running you will have an indicator like
Finished job 4.
1 of 31 steps (3%) done
.
.
.
Finished job 5.
10 of 29 steps (34%) done

In the end you will see a success message.


Please keep me updated.

~Best,
Medhat

**Comment by NGLJ** [comment]
Is  because of "sample_extension" is set to "gz"?

# Information about samples
###########################
sample_directory: "" # samples by default should exist in this directory unless otherwise.
read_type: "ont" # ont,clr or ccs
sample_extension: "gz" # please add sample extension ex: fastq , fasta or gz # defaul gz
# sample list If specified this will overlap the default behavior which is all samples in directory specified above.
# If you leave empty the input will be all the samples in directory.
sample_list: []
###########################
best,
XFY

**Comment by cindylanzao-design** [comment]
Hi，MeHelmy

We are trying your software, it is great with abundant analysis result. It runs perfectly on one or two chromosome, however, when we want to expand our analysis item to more chromosome, that is from chr1 to chr4,  we got following error message. It also fails, If we expand the analysis item to whole genome, that is from chr1 to chr22, plus chrX and chrY, and the error message goes the same. We guess it may because of the shortage of computing resource, however, we are not sure. Could you please help to figure it out? Thanks very much!

The attachment is the configuration file and below is the screenshot of the error:
[cid:image001.png@01D5E5D8.DF6EE0E0]


Best,
Xinyue
发件人: Medhat [mailto:notifications@github.com]
发送时间: 2020年2月12日 5:33
收件人: MeHelmy/princess <princess@noreply.github.com>
抄送: lixinyue (H) <lixinyue12@huawei.com>; Author <author@noreply.github.com>
主题: Re: [MeHelmy/princess] can your software be used on a local machine (#3)


First I would like you to update Princess git pull, then you can see the -e parameter.

I tested Princess on Linux machine with 40G of RAMs and it runs fine (I used only chromosomes 22).

For config.yaml file:

The reference must be indexed, to do that use samtools faidx your_refernce.fai.

Then let us go to the config.yaml to add the required information:
under the title Information about reference we will update these fields:

In this field ref: [] add your reference version either 37 or 38: e.g ref: ['37']

In references: add the location of the reference: e.g '37': "/home/user/somdir/genome.fa"

AS we are just testing we will use chromosome 22 (you can use what you find interesting for you)
to know what your chromosomes named like take a look at the first column in the indexed genome in our case it will be /home/user/somdir/genome.fa.fai

chr_list: '37': ['22'] # leave free if you wish to call all chromosomes ex: '37': [].

So in the end, it will be:

ref: ['37']

references:

  '37': "/home/user/somdir/genome.fa"

  '38': "

chr_list:

  '37': ['22']

  '38': []



Now, let us say that you have a fasta file in /home/user/somdir/sample.fasta

To test what shall run use:
princess_location/princess/princess -c all -s /home/user/somdir/sample.fasta -d ./analysis_37 -r ont -e -npr

-c    The command that will run

-s    One or more input file

-d   Where the output should go

-r    Data type

-e  Do not use the cluster configuration

Results will be something like:

Building DAG of jobs...
Job counts:
count jobs
1 SV_SNP_compained
1 add_rg
1 all
1 all_phased
1 bam_2_tab
1 bam_stat
2 bgzip_file
2 call_snps_chunk
1 change_sample_name
1 concat_chromosome
1 gt
4 index_bam
1 merge_align
2 minimap2
1 partion_bam
1 phase_sv
1 phasing
1 reads_stat
1 sniffles
1 snp_stat
1 stat
1 sv_stat
2 vcf_index
1 vcf_sort
31 # this is the number of jobs that will run, it will change if we used more samples.

If you take a look to the analysis_37 directory you will find new directories were created, Besides symbolic links to the input fasta files

bin/
envs/
modules/
pictures/
scripts/

Time to do the real run.

My input file is a fastq file for chromosome 22
Reads: 173298
Bases: 681069807
Mean read length: 3930.0500121178547
Median: 1984.0
Max: 754862
Min: 173
N50: 7643

princess/princess -c all -s minimap_37_chr22.fastq -d ./analysis_37 -r ont -e

Building DAG of jobs...
Creating conda environment envs/princess_env.yaml...
Downloading and installing remote packages.
Environment for envs/princess_env.yaml created (location: .snakemake/conda/b929bbb8) # Name of the conda environment will be different, and you may need to delete after you finish your analysis.
Creating conda environment envs/clair_env.yaml...
Downloading and installing remote packages.

This will result in (In your case will be different sample name)

Job 15: Running minimap2 , sample is: minimap_37_chr22.fastq

for each job running you will have an indicator like
Finished job 4.
1 of 31 steps (3%) done
.
.
.
Finished job 5.
10 of 29 steps (34%) done

In the end you will see a success message.

Please keep me updated.

~Best,
Medhat

**Comment by MeHelmy** [comment]
Thank you,

Normally, Princess was developed to run on big data, which requires an abundance of computer power (depending on the tool used for the analysis), that is why at first I suggested chromosome or two, but for more data, you may need to use a cluster or a more powerful computer.
I can not see the error from your message, please re-post it.

~Best,
medhat

**Comment by cindylanzao-design** [comment]
Hi, Medhat

Thanks, I did not post the error on the github. How about a CPU based computer with 16 core and over 64 GB memory, can this satisfy the requirement of running princess on a whole genome？

Best，
Xinyue

发件人: Medhat [mailto:notifications@github.com]
发送时间: 2020年2月18日 0:22
收件人: MeHelmy/princess <princess@noreply.github.com>
抄送: lixinyue (H) <lixinyue12@huawei.com>; Author <author@noreply.github.com>
主题: Re: [MeHelmy/princess] can your software be used on a local machine (#3)


Thank you,

Normally, Princess was developed to run on big data, which requires an abundance of computer power (depending on the tool used for the analysis), that is why at first I suggested chromosome or two, but for more data, you may need to use a cluster or a more powerful computer.
I can not see the error from your message, please re-post it.

~Best,
medhat

**Comment by MeHelmy** [comment]
It should work (remember that coverage also plays a role.), but please if there is an error send it to me, so I can help more, Or make it possible.

Edit:
If you managed to run Princess successfully even on one chromosome, there is a directory called `benchmark`, that contains computer usage, e.g:
~~~
s       h:m:s   max_rss max_vms max_uss max_pss io_in   io_out  mean_load
6106.5792       1:41:46 15334.64        17829.96        15247.62        15286.01        3.38    0.00    415.87
~~~
RSS (Resident Set Size)
VMS (Virtual Memory Size)
PSS (proportional set size)
USS (unique set size)

For example, the above process was using exactly  15247.62 MB = 15 GB.

All the memory usage and IO counts are reported in MB.
Thanks,
Medhat

**Comment by cindylanzao-design** [comment]
Hi，Medhat

Here is the log file, this is a 8 core, 64 GB memory GPU machine. Could you please help to figure it out?

Thanks!

Best,
Xinyue

发件人: Medhat [mailto:notifications@github.com]
发送时间: 2020年2月18日 3:26
收件人: MeHelmy/princess <princess@noreply.github.com>
抄送: lixinyue (H) <lixinyue12@huawei.com>; Author <author@noreply.github.com>
主题: Re: [MeHelmy/princess] can your software be used on a local machine (#3)


It should work, but please if there is an error send it to me, so I can help more, Or make it possible.

Thanks,
Medhat

**Comment by MeHelmy** [comment]
> Hi，Medhat Here is the log file, this is a 8 core, 64 GB memory GPU machine. Could you please help to figure it out? Thanks! Best

Where is the log file?

**Comment by cindylanzao-design** [comment]
Sorry, forget the attachment.


发件人: Medhat [mailto:notifications@github.com]
发送时间: 2020年2月18日 10:48
收件人: MeHelmy/princess <princess@noreply.github.com>
抄送: lixinyue (H) <lixinyue12@huawei.com>; Author <author@noreply.github.com>
主题: Re: [MeHelmy/princess] can your software be used on a local machine (#3)


Hi，Medhat Here is the log file, this is a 8 core, 64 GB memory GPU machine. Could you please help to figure it out? Thanks! Best

Where is the log file?

**Comment by MeHelmy** [comment]
> Sorry, forget the attachment. 

No problem, but it is still missing.

**Comment by cindylanzao-design** [comment]
There is seems a restriction on the file I should send out, thus, I copy & paste the log file below:
Thanks


Finished job 46.
10 of 69 steps (14%) done
[Tue Feb 18 08:49:11 2020]
Finished job 44.
11 of 69 steps (16%) done
[Tue Feb 18 08:51:40 2020]
Finished job 41.
12 of 69 steps (17%) done
[Tue Feb 18 08:53:53 2020]
Finished job 49.
13 of 69 steps (19%) done
[Tue Feb 18 08:57:16 2020]
Error in rule call_snps_chunk:
    jobid: 52
    output: /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr2_8.vcf
    conda-env: /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/conda_env/6fcbc117
    shell:

        export PATH=$PWD/bin/pypy/bin:$PATH &&         clair.py callVarBam             --chkpnt_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/bin/modules/ont/model             --bam_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/align/minimap/data.bam             --ref_fn /mnt/data1/lixinyue/data/hg19.fa             --minCoverage 2             --ctgName chr2             --ctgStart 216177216             --ctgEnd 243199373             --threads 5 --call_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr2_8.vcf

        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job call_snps_chunk since they might be corrupted:
/mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr2_8.vcf
[Tue Feb 18 08:59:54 2020]
Finished job 39.
14 of 69 steps (20%) done
[Tue Feb 18 09:05:27 2020]
Finished job 35.
15 of 69 steps (22%) done
[Tue Feb 18 09:09:54 2020]
Error in rule call_snps_chunk:
    jobid: 47
    output: /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr2_3.vcf
    conda-env: /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/conda_env/6fcbc117
    shell:

        export PATH=$PWD/bin/pypy/bin:$PATH &&         clair.py callVarBam             --chkpnt_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/bin/modules/ont/model             --bam_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/align/minimap/data.bam             --ref_fn /mnt/data1/lixinyue/data/hg19.fa             --minCoverage 2             --ctgName chr2             --ctgStart 81066456             --ctgEnd 108088608             --threads 5 --call_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr2_3.vcf

        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job call_snps_chunk since they might be corrupted:
/mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr2_3.vcf
[Tue Feb 18 09:16:37 2020]
Error in rule call_snps_chunk:
    jobid: 56
    output: /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr3_3.vcf
    conda-env: /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/conda_env/6fcbc117
    shell:

        export PATH=$PWD/bin/pypy/bin:$PATH &&         clair.py callVarBam             --chkpnt_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/bin/modules/ont/model             --bam_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/align/minimap/data.bam             --ref_fn /mnt/data1/lixinyue/data/hg19.fa             --minCoverage 2             --ctgName chr3             --ctgStart 84866754             --ctgEnd 113155672             --threads 5 --call_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr3_3.vcf

        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job call_snps_chunk since they might be corrupted:
/mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr3_3.vcf
[Tue Feb 18 09:21:25 2020]
Error in rule call_snps_chunk:
    jobid: 59
    output: /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr3_6.vcf
    conda-env: /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/conda_env/6fcbc117
    shell:

        export PATH=$PWD/bin/pypy/bin:$PATH &&         clair.py callVarBam             --chkpnt_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/bin/modules/ont/model             --bam_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/align/minimap/data.bam             --ref_fn /mnt/data1/lixinyue/data/hg19.fa             --minCoverage 2             --ctgName chr3             --ctgStart 169733508             --ctgEnd 198022430             --threads 5 --call_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr3_6.vcf

        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job call_snps_chunk since they might be corrupted:
/mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr3_6.vcf
[Tue Feb 18 09:26:03 2020]
Error in rule call_snps_chunk:
    jobid: 54
    output: /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr3_1.vcf
    conda-env: /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/conda_env/6fcbc117
    shell:

        export PATH=$PWD/bin/pypy/bin:$PATH &&         clair.py callVarBam             --chkpnt_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/bin/modules/ont/model             --bam_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/align/minimap/data.bam             --ref_fn /mnt/data1/lixinyue/data/hg19.fa             --minCoverage 2             --ctgName chr3             --ctgStart 28288918             --ctgEnd 56577836             --threads 5 --call_fn /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr3_1.vcf

        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job call_snps_chunk since they might be corrupted:
/mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snp/minimap/data.split.chr3_1.vcf
[Tue Feb 18 09:26:49 2020]
Finished job 36.
16 of 69 steps (23%) done
[Tue Feb 18 09:27:54 2020]
Finished job 38.
17 of 69 steps (25%) done
[Tue Feb 18 09:31:13 2020]
Finished job 64.
18 of 69 steps (26%) done
[Tue Feb 18 09:32:14 2020]
Finished job 51.
19 of 69 steps (28%) done
[Tue Feb 18 09:33:23 2020]
Finished job 63.
20 of 69 steps (29%) done
[Tue Feb 18 09:33:59 2020]
Finished job 50.
21 of 69 steps (30%) done
[Tue Feb 18 09:35:43 2020]
Finished job 65.
22 of 69 steps (32%) done
[Tue Feb 18 09:35:44 2020]
Finished job 60.
23 of 69 steps (33%) done
[Tue Feb 18 09:35:53 2020]
Finished job 43.
24 of 69 steps (35%) done
[Tue Feb 18 09:36:26 2020]
Finished job 53.
25 of 69 steps (36%) done
[Tue Feb 18 09:37:10 2020]
Finished job 42.
26 of 69 steps (38%) done
[Tue Feb 18 09:37:17 2020]
Finished job 11.
27 of 69 steps (39%) done
[Tue Feb 18 09:37:47 2020]
Finished job 57.
28 of 69 steps (41%) done
[Tue Feb 18 09:38:00 2020]
Finished job 55.
29 of 69 steps (42%) done
[Tue Feb 18 09:38:05 2020]
Finished job 45.
30 of 69 steps (43%) done
[Tue Feb 18 09:38:08 2020]
Finished job 62.
31 of 69 steps (45%) done
[Tue Feb 18 09:38:08 2020]
Finished job 37.
32 of 69 steps (46%) done
[Tue Feb 18 09:38:25 2020]
Finished job 48.
33 of 69 steps (48%) done
[Tue Feb 18 09:38:38 2020]
Finished job 58.
34 of 69 steps (49%) done
[Tue Feb 18 09:38:44 2020]
Finished job 34.
35 of 69 steps (51%) done
[Tue Feb 18 09:38:49 2020]
Finished job 61.
36 of 69 steps (52%) done
[Tue Feb 18 09:38:53 2020]
Finished job 66.
37 of 69 steps (54%) done
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
mkdir -p /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snake_log &&    find . -maxdepth 1 -name 'snakejob.*' -type f -print0 | xargs -0r mv -t /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snake_log &&    cat pictures/fail.txt
Complete log: /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/.snakemake/log/2020-02-18T081742.852960.snakemake.log
发件人: Medhat [mailto:notifications@github.com]
发送时间: 2020年2月18日 10:56
收件人: MeHelmy/princess <princess@noreply.github.com>
抄送: lixinyue (H) <lixinyue12@huawei.com>; Author <author@noreply.github.com>
主题: Re: [MeHelmy/princess] can your software be used on a local machine (#3)


Sorry, forget the attachment.

No problem, but it is still missing.

**Comment by MeHelmy** [comment]
Thanks,
The error happened while calling SNPs (it may not be a memory issue.).
Can you please look for an error file contains `call_snps_chunk` in its name in directory ` /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snake_log ` and copy its content here.

~Best,
Medhat

**Comment by cindylanzao-design** [comment]
There is no file in the directory you mentioned in the email. The only log it the one I copied earlier in the email.

Best,
Xinyue

发件人: Medhat [mailto:notifications@github.com]
发送时间: 2020年2月18日 11:05
收件人: MeHelmy/princess <princess@noreply.github.com>
抄送: lixinyue (H) <lixinyue12@huawei.com>; Author <author@noreply.github.com>
主题: Re: [MeHelmy/princess] can your software be used on a local machine (#3)


Thanks,
The error happened while calling SNPs (it may not be a memory issue.).
Can you please look for an error file contains call_snps_chunk in its name in directory /mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/snake_log and copy its content here.

~Best,
Medhat

**Comment by MeHelmy** [comment]
Is this file containt the same information?

`/mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/.snakemake/log/2020-02-18T081742.852960.snakemake.log`

**Comment by cindylanzao-design** [comment]
Thanks very much for your relpy. Yes, that is what I copied in the earlier email.


Best,
Xinyue

发件人: Medhat [mailto:notifications@github.com]
发送时间: 2020年2月18日 11:41
收件人: MeHelmy/princess <princess@noreply.github.com>
抄送: lixinyue (H) <lixinyue12@huawei.com>; Author <author@noreply.github.com>
主题: Re: [MeHelmy/princess] can your software be used on a local machine (#3)


Is this file containt the same information?

/mnt/data1/lixinyue/docker/test_xfy/princess/princess-master/analysis_37/.snakemake/log/2020-02-18T081742.852960.snakemake.log

**Comment by MeHelmy** [comment]
My bad, 
I made a small change, please `git pull` so you can have the changes. Please run Princess again, it will begin from where it finished last time, thus it will only try to call SNPs, then we could have a better look at what happened. 

Thanks,
Medhat

**Comment by cindylanzao-design** [comment]
What kind of new character do you add to it? If it is run from where it stopped, I do not think we really need this character, when our customer  rerun the program, we assume they will want to overwrite the previous file. Thanks!

Best,
Xinyue

发件人: Medhat [mailto:notifications@github.com]
发送时间: 2020年2月18日 11:52
收件人: MeHelmy/princess <princess@noreply.github.com>
抄送: lixinyue (H) <lixinyue12@huawei.com>; Author <author@noreply.github.com>
主题: Re: [MeHelmy/princess] can your software be used on a local machine (#3)


My bad,
I made a small change, please git pull so you can have the changes. Please run Princess again, it will begin from where it finished last time, thus it will only try to call SNPs, then we could have a better look at what happened.

Thanks,
Medhat

**Comment by MeHelmy** [comment]
This is the default behavior in Princess. It runs from where it ended, if you gave it the same data; unless you removed the old result directories it will run from the start.

**Comment by cindylanzao-design** [comment]
Currently,  we have ran it successfully in a local machine, and is test it on the cloud, will keep you update if any mistakes occur.

Thanks!

发件人: Medhat [mailto:notifications@github.com]
发送时间: 2020年2月18日 23:36
收件人: MeHelmy/princess <princess@noreply.github.com>
抄送: lixinyue (H) <lixinyue12@huawei.com>; Author <author@noreply.github.com>
主题: Re: [MeHelmy/princess] can your software be used on a local machine (#3)


This is the default behavior in Princess. It runs from where it ended, if you gave it the same data; unless you removed the old result directories it will run from the start.
